### ISSUE TITLE
Morale "Stylish" is inconsistent.

### ISSUE THREAD

**Issue by Tommy-os** [CleverRaven/Cataclysm-DDA#48495]
# Describe the bug

 DEBUG    : Morale "Stylish" is inconsistent.

 FUNCTION : player_morale::consistent_with(const player_morale&) const::<lambda(const player_morale&, const player_morale&)>
 FILE     : src/morale.cpp
 LINE     : 790

After this error, all non permanent morale bonuses are reset (such as from food).

# Steps To Reproduce

1. Put 2 silver watches into your inventory
2. Equip both silver watches
3. Switch the second silver watch onto a different hand (c key)
4. Drop both silver watches onto ground at the same time

Close inventory, wait one turn and reopen inventory after each step if you fail to reproduce the error.

# Expected behavior

No error, preserved temporary from other sources morale.

# Screenshots

# Versions and configuration

# Additional context

**Comment by OliveChukar** [comment]
Tested on 0.E-10395-g4525b4f [64-bit] with Dark Days Ahead [dda] only no mods.  I tested with one gold and one platinum watch and got the same error message. All morale positive and negative except optimist was removed by the error.

**Comment by Xpyder** [comment]
This happened for me with the removal of FANCY from a lot of items, but waiting 10 turns temporary bonuses came back.

I will say I looked at the .cpp and there may be a logic error, in that the only place `consistent_with` is checked is immediately followed with a fix consistency command, but they use different debug message outputs.

i.e.
morale.cpp:790 
    debugmsg( "Morale \"%s\" is inconsistent.", lhp.get_name() );
might need to be changed to
    add_msg_debug( "Morale \"%s\" is inconsistent.", lhp.get_name() );
to match character.cpp:10890
    add_msg_debug( "%s morale was recovered.", disp_name( true ) );

But I don't actually know the difference between the debug commands so not positive this is the best change

**Comment by Xpyder** [comment]
Also character.cpp:10774 might need to be reviewed
`*morale = player_morale( test_morale ); // Recover consistency`

**Comment by Xpyder** [comment]
and for posterity here's a commented version of the consistent_with code explaining what I think it seems to be doing
```
bool player_morale::consistent_with( const player_morale &morale ) const
{
    // For all non-permenant morale points
    // Make sure that all points in left_hand_side also exist in right_hand_side
    // and that their net_bonus() match
    const auto test_points = []( const player_morale & lhs, const player_morale & rhs ) {
        // For each point in left_hand_side.points
            // if point is permanent, skip
        for( const auto &lhp : lhs.points ) {
            if( !lhp.is_permanent() ) {
                continue;
            }

            // Find point in right_hand_side that matches the left_hand_point
            const auto iter = std::find_if( rhs.points.begin(), rhs.points.end(),
            [ &lhp ]( const morale_point & rhp ) {
                return lhp.matches( rhp );
            } );

            if( 
                // If there's no matching point in the left_hand_side vs right_hand_side
                iter == rhs.points.end() 
                || 
                // If if the net bonuses don't match between left_hand_side vs right_hand_side
                lhp.get_net_bonus() != iter->get_net_bonus() 
            ) {
                debugmsg( "Morale \"%s\" is inconsistent.", lhp.get_name() );
                return false;
            }
        }

        return true;
    };
```

**Comment by Xpyder** [comment]
I suspect what's happening is the code that checks for consistency happens every 9 turns, and if you happen to have changed your possible morale right at that instant (by dropping something fancy while 'stylish') it can generate a transient error that gets auto-corrected later on

Can I get confirmation if the morale bonuses correct themselves after about 1 in-game minute?

**Comment by OliveChukar** [comment]
I just tested on (cdda-windows-tiles-x64-2021-06-20-0737) and the temporary morale bonuses did not reappear.  I checked every few seconds for 5 minutes.

**Comment by Xpyder** [comment]
Interesting, I know that in my case I definitely received a "got comfy" and food morale bonuses within a minute or two, but I didn't do extensive testing to determine the limitations (other than checking if any of my existing clothing still had the 'fancy' tag).

I assume you used the testing method listed in the OP?  If not are you able to provide a save we can test out?

**Comment by RenechCDDA** [comment]
Still occurs as of https://github.com/CleverRaven/Cataclysm-DDA/commit/c94dc0387513d5bba49eb7d74b2b51a7a4d8c4a0

This specific check has given me some trouble before, so ideally it just gets refactored out one day.

**Comment by VladMasarik** [comment]
Not sure if this is the same case but I got `player_morale::took_prozac is inconsistent` errors, and I have a save that seems to have the issue always replicable, let me know if you would need the save file. I am reporting here because I had the stylish error pop up at some point as well, but was not able to replicate it.

**Comment by ZhilkinSerg** [comment]
No longer relevant after #79745
